packages/core/src/api/clipboard/fromClipboard/pasteExtension.ts

```diff
@@ -61,15 +64,37 @@ export const createPasteFromClipboardExtension = <
                   return true;
                 }
 
+                if (format === "text/markdown") {
+                  markdownToHTML(data).then((html) => {
+                    view.pasteHTML(html);
+                  });
+                  return true;
+                }
+
                 if (format === "text/html") {
+                  if (editor.settings.pasteBehavior === "prefer-markdown") {
+                    // Use plain text instead of HTML if it looks like Markdown
+                    const plainText =
+                      event.clipboardData!.getData("text/plain");
+
+                    if (is(plainText)) {
+                      // Convert Markdown to HTML first, then paste as HTML
+                      markdownToHTML(plainText).then((html) => {
+                        view.pasteHTML(html);
+                      });
+                      return true;
+                    }
+                  }
                   const htmlNode = nestedListsToBlockNoteStructure(data.trim());
```

Comment:
Might be worth checking whether we need this for the other cases as well (`nestedListsToBlockNoteStructure`)